pkg/cfn/manager/iam.go

```diff
@@ -31,7 +34,22 @@ func (c *StackCollection) createIAMServiceAccountTask(errs chan error, spec *api
 	}
 	spec.Tags[api.IAMServiceAccountNameTag] = spec.NameString()
 
-	return c.CreateStack(name, stack, spec.Tags, nil, errs)
+	if err := c.CreateStack(name, stack, spec.Tags, nil, errs); err != nil {
+		if !replaceExistingRole {
+			return err
+		}
+		var awsErr awserr.Error
+		if errors.As(err, &awsErr) && awsErr.Code() == cfn.ErrCodeAlreadyExistsException {
+			logger.Debug("CFN stack for IRSA already exists, replacing it with a new stack")
+			if err := c.DeleteStackByNameSync(name); err != nil {
+				close(errs)
+				return errors.Wrap(err, "error deleting stack")
+			}
+			return c.createIAMServiceAccountTask(errs, spec, oidc, false)
```

Comment:
Am I right in thinking that VPC controller installation only occurs during cluster creation? If so then when will the scenario occur where the SA already exists occur?

Reply:
>If so then when will the scenario occur where the SA already exists occur?

The VPC controller can be installed on an existing cluster using  `eksctl utils install-vpc-controllers`.

Reply:
So is the scenario here when a user runs `eksctl utils install-vpc-controllers` after a cluster installation that already installed the VPC controller? If so why do we need to delete & recreate, can't just check for its existence and exit early if it does?

Reply:
This is to allow applying any changes to the policy document. If we just check for existence, any changes made to the policy document in code will not be applied.

Reply:
but is it needed for this PR? I'm just trying to understand when this codepath is used in relation to `vpc-controller` 😄 

It might be that this isn't needed as we will add the ability to update IAM policies without deleting here https://github.com/weaveworks/eksctl/pull/3064

Reply:
> I'm just trying to understand when this codepath is used in relation to vpc-controller

This codepath is invoked when `utils install-vpc-controllers` is run on a cluster that already has the VPC controller installed (either as part of `create cluster` or with `utils install-vpc-controllers` itself). The ServiceAccount created for the role needs the role ARN, so if we skip creation of this stack if it already exists, we'll have to fetch it and get the role ARN, and any changes to the policy document will be ignored.

Reply:
Ah I see, so is this just for the scenario in which: I have an already created eksctl cluster with oidc enabled and the VPC controller installed, but its missing the policies because I created it with an older version of eksctl? If so that makes sense 😄 

Apart from that scenario you'd never need to run `utils install-vpc-controllers` if its already installed correct?

Reply:
>Ah I see, so is this just for the scenario in which: I have an already created eksctl cluster with oidc enabled and the VPC controller installed, but its missing the policies because I created it with an older version of eksctl?

That's right.

>Apart from that scenario you'd never need to run utils install-vpc-controllers if its already installed correct?

Users may also run it to upgrade to a newer version of the VPC controller.

Reply:
Thanks! Makes sense. Does this add more downtime for users when they update the VPC controller as we will always delete and recreate the policies even if they are unchanged? Very minor, but it could be removed by using the update functionality https://github.com/weaveworks/eksctl/pull/3064

Reply:
>Does this add more downtime for users when they update the VPC controller as we will always delete and recreate the policies even if they are unchanged?

Yes. 

>Very minor, but it could be removed by using the update functionality #3064

Yeah, the ideal way would be to update the policies in the existing stack, but since we didn't already have code that updates the policies for IRSA, I went with recreating it. I was aware of your PR but it wasn't in `master` when I was working on this PR. After your PR is merged, I can change this code to use the update feature (or it can be done as an improvement later).